configure.d/1_blk_queue_exit_workaround_for_rhel.conf

```diff
@@ -6,14 +6,51 @@
 
 . $(dirname $3)/conf_framework
 
-# RHEL8.5 kernel 4.18 subversion targeted for the workaround is 348.7.1.
-# The workaround is needed starting this version and up.
-QUEUE_PUT_WA_LINUX_MAJOR_SINCE=4
-QUEUE_PUT_WA_LINUX_MINOR_SINCE=18
-QUEUE_PUT_WA_LINUX_SUB_SINCE=0
-QUEUE_PUT_WA_RHEL_MAJOR_SINCE=348
-QUEUE_PUT_WA_RHEL_MINOR_SINCE=7
-QUEUE_PUT_WA_RHEL_SUB_SINCE=1
+# Check whether version A is the same as (or newer than) version B.
+#
+# Version consists of three elements: major, minor and sub.
+# The last argument (GT_LVL) tells at which level it is acceptable
+# for version A to be newer than B. It can be "major", "minor", "sub"
+# or unspecified.
+# - For GT_LVL="major" used, then all versions that are newer than B
+#   are accepted.
+# - For GT=LVL="minor" only versions that are newer and have the same
+#   major number.
+# - For GT_LVL="sub" only version that are newer and have the same major
+#   and minor number are accepted.
+# - For GT_LVL unspecified version A needs to be exactly the same as
+#   version B.
+compare_version () {
+	MAJOR_A=$1
+	MINOR_A=$2
+	SUB_A=$3
+	MAJOR_B=$4
+	MINOR_B=$5
+	SUB_B=$6
+	GT_LVL=$7
+	if   [ "$MAJOR_A"  -eq "$MAJOR_B" ]; then
+		if    [ "$MINOR_A" -eq "$MINOR_B" ]; then
+			if    [ "$SUB_A" -eq "$SUB_B" ]; then
+				return 0
+			elif  [ -n "$GT_LVL" ] && [ "$SUB_A" -gt "$SUB_B" ]; then
+				return 0
+			fi
+		elif  [ "$GT_LVL" = "major" -o "$GT_LVL" = "minor" ] &&
+			  [ "$MINOR_A" -gt "$MINOR_B" ]; then
+			return 0
+		fi
+	elif  [ "$GT_LVL" = "major" ] && [ "$MAJOR_A" -gt "$MAJOR_B" ]; then
+		return 0
+	fi
+	return 1
+}
+
+compare_kernel_version () {
```

Comment:
maybe add the 4-th parameter (GT_LEVEL) so that the interface is the same as compare_rhel_kernel_version?